core-rust/core-api-server/src/core_api/conversions/numerics.rs

```diff
@@ -150,37 +150,30 @@ pub fn to_unix_timestamp_ms(time: std::time::SystemTime) -> Result<i64, MappingE
     })
 }
 
-pub fn to_api_instant(instant: &Instant) -> Result<models::Instant, MappingError> {
-    to_api_instant_from_safe_timestamp(instant.seconds_since_unix_epoch.checked_mul(1000).ok_or(
-        MappingError::IntegerError {
-            message: "Timestamp must be representable as millis in i64".to_owned(),
-        },
-    )?)
+pub fn to_api_scrypto_instant(instant: &Instant) -> Result<models::ScryptoInstant, MappingError> {
+    let timestamp_seconds = instant.seconds_since_unix_epoch;
+    let date_time = NaiveDateTime::from_timestamp_opt(timestamp_seconds, 0);
+    Ok(models::ScryptoInstant {
+        unix_timestamp_seconds: to_api_i64_as_string(timestamp_seconds),
+        date_time: date_time.map(to_canonical_rfc3339_string),
+    })
 }
 
-pub fn to_api_instant_from_safe_timestamp(
-    timestamp_millis: i64,
-) -> Result<models::Instant, MappingError> {
-    if !(MIN_API_TIMESTAMP_MS..=MAX_API_TIMESTAMP_MS).contains(&timestamp_millis) {
-        return Err(MappingError::IntegerError {
-            message: format!("Timestamp ms must be >= 0 and <= {MAX_API_TIMESTAMP_MS}"),
-        });
-    }
-    use chrono::prelude::*;
-    let date_time = NaiveDateTime::from_timestamp_millis(timestamp_millis)
-        .map(|d| {
-            DateTime::<Utc>::from_utc(d, Utc).to_rfc3339_opts(chrono::SecondsFormat::Millis, true)
-        })
-        .ok_or_else(|| MappingError::IntegerError {
-            message: "Timestamp invalid when converted to DateTime".to_string(),
-        })?;
-
-    Ok(models::Instant {
-        unix_timestamp_ms: timestamp_millis,
-        date_time,
+pub fn to_api_clamped_instant(timestamp_millis: i64) -> Result<models::InstantMs, MappingError> {
+    let clamped_timestamp_millis =
+        timestamp_millis.clamp(MIN_API_TIMESTAMP_MS, MAX_API_TIMESTAMP_MS);
+    let date_time = NaiveDateTime::from_timestamp_millis(clamped_timestamp_millis)
+        .expect("it just got clamped to 100% supported range above");
+    Ok(models::InstantMs {
+        unix_timestamp_ms: clamped_timestamp_millis,
+        date_time: to_canonical_rfc3339_string(date_time),
```

Comment:
Can we add some unit tests for the extremes of [0, 100000000000000] (both valid and not-valid)?